fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryWrapper.java

```diff
@@ -149,6 +143,15 @@ public AnalyzingInfixSuggester getAutocompleteSuggester(@Nonnull LuceneAnalyzerW
         return suggester;
     }
 
+    private boolean suggesterRenewNeeded(@Nonnull LuceneAnalyzerWrapper indexAnalyzerWrapper,
+                                         @Nonnull LuceneAnalyzerWrapper queryAnalyzerWrapper,
+                                         @Nullable IndexOptions indexOptions) {
+        return suggester == null
+               || !suggesterIndexAnalyzerId.equals(indexAnalyzerWrapper.getUniqueIdentifier())
+               || !suggesterQueryAnalyzerId.equals(queryAnalyzerWrapper.getUniqueIdentifier())
+               || indexOptions != null && !indexOptions.equals(suggesterFieldIndexOptions);
+    }
```

Comment:
This required me to remember the order of boolean evaluation.  I think this is right but a parenthesis might make it easier to understand.

Reply:
Added parenthesis